### DISCUSSION TITLE
Tesla API 408 error

### DISCUSSION THREAD

**Discussion by markuswells** [teslamate-org/teslamate#3577] [Getting started]
I have still not been able to get data, but teslamate has the following error:

> 2023-12-19 16:56:34.282 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1493003479502931/vehicle_data -> 408 (9158.482 ms)
2023-12-19 16:56:34.282 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
2023-12-19 16:56:34.282 car_id=1 [error] Error / :unknown

I dont see how its my firewall, but I am also fairly stumped. Any help is appreciated

**Comment by cwanja** [comment]
You suggest firewall is not the issue. Why do you think that is not the issue? If you have not be able to get any data, it seems like a network issue preventing the app from talking to Tesla. 

Have you tried adjusting your network settings? How do you have TeslaMate installed?

**Comment by brianmay** [comment]
408 errors are normal if the car is asleep or otherwise unreachable by the Tesla API.

If this is the only error you are seeing, maybe check that the car is awake and has a good network connection.

**Comment by markuswells** [comment]
My tesla does not have premium connectivity, so does it need access to wifi, or does the API use the default tesla connection?
Also
I have port forwarding on my router setup for 4000, 3000, 1883, 5432 & traefik running proxy in front of the teslaMate docker- and since I can see teslaMate and Grafana I am not sure what other firewall changes I need to do

**Reply by brianmay** [reply]
My understanding is that lack of premium connectivity should not affect the ability of the API to access the car. It only affects user facing services that are considered "premium". <cynical>Because Tesla really wants the data from your car.</cynical>

My car is an older one that has lifetime premium connectivity, so I can't say this with 100% confidence.

You should be able to confirm, if the mobile app works, then you have API access to the car. Maybe try outside Bluetooth range just in case it is using that (assuming you configured this).

**Comment by markuswells** [comment]
I am still not getting data into the dashboard. I have connected the car via wifi, so either way the car is accessible and as mentioned the app works fine.
So the car is awake and accessible, so what would stop TeslaMate and the API from being able to communicate?
